lib/i2c.ex

```diff
@@ -31,7 +35,7 @@ defmodule Circuits.I2C do
   On success, this returns a reference to the I2C bus resource.  Use the
   reference in subsequent calls to read/write I2C device
   """
-  @spec open(binary) :: {:ok, reference} | {:error, term}
+  @spec open(i2c_device :: binary) :: {:ok, reference} | {:error, term}
```

Comment:
<img width="676" alt="screen shot 2018-11-04 at 7 04 42 pm" src="https://user-images.githubusercontent.com/64669/47971914-8b1bb380-e064-11e8-8216-66153a6d6049.png">

That's what the docs look like. If I get the gist of what you want to achieve, how do you feel about renaming the parameter (not the type, but the actual parameter to make it more clear). Also, I was thinking that the reference returned deserved a type of its own. Plus parentheses to make them consistent with ex_doc. How's this:

```elixir
@spec open(String.t()) :: {:ok, i2c_device()} | {:error, any()}
def open(device_name) do
...
```

Reply:
I know this parentheses thing is going to cause pain. My feeling is that the battle has been lost. ex_doc does parentheses, the Visual Studio Code plugin (and I assume others) puts in parentheses by default. I realize there are exceptions in the Elixir code base and if you bring this up with José and he says "no parentheses" is the Elixir convention and ex_doc is wrong, then I'll be good with no parentheses.

Reply:
I thought about this some more. What I wrote sounds like a higher bar than I intended. I'm fine with punting for now, but we should file an issue on the Elixir project on parentheses vs. no parentheses to confirm or refute the direction that ex_doc has taken.

Reply:
I can update, I don't mind.  I like that reference being a type of its own actually, I did not think of that.